src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsauditapi/services/AuditS3Client.kt

```diff
@@ -33,19 +36,40 @@ class AuditS3Client(
       .contentMD5(md5Base64)
       .build()
 
-    s3Client.putObject(putObjectRequest, RequestBody.fromBytes(jsonBytes))
+    s3Client.putObject(putObjectRequest, RequestBody.fromBytes(parquetBytes))
   }
 
-  private fun generateFilename(auditEvent: HMPPSAuditListener.AuditEvent, auditEventJsonString: String): String {
+  private fun generateFilename(auditEvent: HMPPSAuditListener.AuditEvent): String {
     val whenDateTime = auditEvent.`when`.atZone(ZoneId.systemDefault()).toLocalDateTime()
     return "year=${whenDateTime.year}/month=${whenDateTime.monthValue}/day=${whenDateTime.dayOfMonth}/user=${auditEvent.who}/" +
-      "${getMd5HashFromJsonString(auditEventJsonString)}.json"
+      "${auditEvent.id}.parquet"
   }
 
-  private fun getMd5HashFromJsonString(jsonString: String): String {
-    val bytes = jsonString.toByteArray()
-    val md = MessageDigest.getInstance("MD5")
-    val digest = md.digest(bytes)
-    return digest.joinToString("") { "%02x".format(it) }
+  fun convertToParquetBytes(auditEvent: HMPPSAuditListener.AuditEvent, filename: String): ByteArray {
+    val schema: Schema = Schema.Parser().parse(javaClass.getResourceAsStream("/audit_event.avsc"))
```

Comment:
Can you parse the schema once on init and then reuse it? It would be more efficient than having to load/parse it per event (Im assuming the schema is a reusable object and doesnt have state tying it to an instance).

Reply:
Yes you're right, can do that